### PR TITLE
db: convert PermStore into a basestore

### PR DESCRIPTION
**Pull request by asdine**
Use the basestore package in PermStore

### CODE REVIEW

File: enterprise/cmd/repo-updater/authz/perms_syncer_test.go [hunk 1]
```diff
@@ -144,7 +144,7 @@ func TestPermsSyncer_syncUserPerms(t *testing.T) {
 		db.Mocks.Repos = db.MockRepos{}
 	}()
 
-	permsStore := edb.NewPermsStore(nil, timeutil.Now)
+	permsStore := edb.Perms(nil, timeutil.Now)
```
Comment (ryanslade) [inline enterprise/cmd/repo-updater/authz/perms_syncer_test.go:147]: Is it safe to base nil db here?

**Review by ryanslade — APPROVED** [review]
LGTM, but I think @unknwon should take a look before merge

File: enterprise/cmd/repo-updater/authz/perms_syncer_test.go (hunk 1 above)
Comment (asdine) [reply enterprise/cmd/repo-updater/authz/perms_syncer_test.go:147]: I think so, IIUC every relevant code path is mocked in this test

**Review by unknwon — APPROVED** [review]
Thanks! Great to see it's finally unified :D 

Also `mockReposLister` reported to be unused in "enterprise/cmd/repo-updater/authz/perms_syncer_test.go", could you also remove it since you're on this file?

**Pull request merged**